pkgs/development/compilers/unison/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv, fetchurl, gnutar }:
+
+stdenv.mkDerivation rec {
+  pname = "unison-code-manager";
+  milestone_id = "M1c";
+  version = "0.0.1-alpha-${milestone_id}";
+
+  src = if (stdenv.isDarwin) then
+    fetchurl {
+      url = "https://github.com/unisonweb/unison/releases/download/release/${milestone_id}/unison-osx.tar.gz";
+      sha256 = "03q02r7qc7ybqz16kmpk2d8l9vx28kaj9x59mlxzi8a4mr0j3vzb";
+    }
+  else
+    fetchurl {
+      url = "https://github.com/unisonweb/unison/releases/download/release/${milestone_id}/unison-linux64.tar.gz";
+      sha256 = "1iwynqnp1i39pyq9wc01x7y22y1qa0rrjlx40jjdgnj23y1r6jk4";
+    };
+
+  # TODO: Use the explicit "dontFoo" options once unpacking support is fixed
+  # to not have "no new directories created" as a fatal condition.
+  dontBuild = true;
+  dontConfigure = true;
+  phases = "installPhase";
```

Comment:
Just do
```
sourceRoot= ".";
```

and this should work. followed by fixing https://github.com/NixOS/nixpkgs/pull/66249/files#r311720855